Ventilator mode/oxygen modality data on consultation page

Jacobjeevan opened the pull request:
## Proposed Changes

- Fixes #8264 
- Fixes #8612 
- Fixes #8611 

<img width="1207" alt="image" src="https://github.com/user-attachments/assets/bd14d419-df3d-44ed-9263-fd6dad94012f">

Take a look at the font styling for the label, can adjust as needed; could do the same for label position as well (check the #8264 thread for examples).

<img width="580" alt="image" src="https://github.com/user-attachments/assets/94c4978e-b718-4708-a1e1-f21fe815cc81">

<img width="580" alt="image" src="https://github.com/user-attachments/assets/674a0bbe-e434-4afc-a369-842d320870bb">

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
  - Enhanced `ConsultationVentilatorTab` with data fetching, loading states, and pagination for ventilator reports.
  - Introduced `VentilatorTable` to display ventilator data with improved formatting and internationalization support.
  - Added vertical markers to `VentilatorPlot` for better data visualization.
  - New test suite for "Patient Doctor Connect" functionality, enhancing coverage of user interactions.
  
- **Improvements**
  - Expanded localization support for oxygen modalities and ventilator modes.
  - Improved error handling and data presentation in various components.

- **Bug Fixes**
  - Adjusted handling of undefined values in `BinaryChronologicalChart` component.

- **Chores**
  - Reorganized import statements across various test files for better readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jacobjeevan committed:
Ventilator Data

- Added ventilator data as table on Ventilator tab
- Added Marker Area to graphs under ventilator tab

Jacobjeevan committed:
MarkerLine and MarkerArea

- To do: cleanup (only keep whichever one is chosen, cleanup commented out code)

Jacobjeevan committed:
MarkLine, rm MarkArea code

- Choosing markLine as that represents the data accurately (when switching to bar graph)

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

Jacobjeevan committed:
Rm markArea import

nihal467 commented:
@Jacobjeevan is the PR ready for testing

Jacobjeevan committed:
added overflow to table/mobile view

Jacobjeevan commented:
> @Jacobjeevan is the PR ready for testing

@nihal467 Yea, finalized on using the line graphs in the call yesterday. There maybe some changes with the label styling, but it's good to go for the most part.

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

Jacobjeevan committed:
render most recent mode/modality on graph

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

samholics commented:
tested,
@nihal467 @Jacobjeevan 
- [ ] "Ventilator Spo2" is still showing up on events module when spo2 updated on vitals section on brief round type also on oxygen support section of Respiratory support. it should be renamed to "Spo2" on events module if its not on ventilator support.
- [ ] still oxygen support is showing under as "ventilator Interface" on events module
<img width="239" alt="image" src="https://github.com/user-attachments/assets/715d9f05-629f-4b2e-a20f-a1a28c6d109c">
 note: oxygen support is not a Ventilator Interface

Jacobjeevan commented:
> tested, @nihal467 @Jacobjeevan
> 
>     * [ ]  "Ventilator Spo2" is still showing up on events module when spo2 updated on vitals section on brief round type also on oxygen support section of Respiratory support. it should be renamed to "Spo2" on events module if its not on ventilator support.
> 
>     * [ ]  still oxygen support is showing under as "ventilator Interface" on events module
> 
> 
> <img alt="image" width="239" src="https://private-user-images.githubusercontent.com/75214902/378345541-715d9f05-629f-4b2e-a20f-a1a28c6d109c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MDE1NjksIm5iZiI6MTcyOTUwMTI2OSwicGF0aCI6Ii83NTIxNDkwMi8zNzgzNDU1NDEtNzE1ZDlmMDUtNjI5Zi00YjJlLWEyMGYtYTFhMjhjNmQxMDljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDA5MDEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiNWRlMThkODFhY2ZlZjlkNDEzYWE1NDhlMjllOTY4ODM1OWQ3ODc1OWEwZDdiOWI4ZGU5NDQ2ODU3YTE5YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XJPliWPs5X9J4uc4CqhL6SnhoC5jLXb2fvcOZdcF6wM"> note: oxygen support is not a Ventilator Interface

I believe I missed the changes when merging the translations; I'll get it fixed 👍  

As far as this goes,

>  still oxygen support is showing under as "ventilator Interface" on events module

Is it acceptable to have "Respiratory support" as the caption for oxygen support as well as NIV/IV?

samholics commented:
> > tested, @nihal467 @Jacobjeevan
> > ```
> > * [ ]  "Ventilator Spo2" is still showing up on events module when spo2 updated on vitals section on brief round type also on oxygen support section of Respiratory support. it should be renamed to "Spo2" on events module if its not on ventilator support.
> > 
> > * [ ]  still oxygen support is showing under as "ventilator Interface" on events module
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > <img alt="image" width="239" src="https://private-user-images.githubusercontent.com/75214902/378345541-715d9f05-629f-4b2e-a20f-a1a28c6d109c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MDE1NjksIm5iZiI6MTcyOTUwMTI2OSwicGF0aCI6Ii83NTIxNDkwMi8zNzgzNDU1NDEtNzE1ZDlmMDUtNjI5Zi00YjJlLWEyMGYtYTFhMjhjNmQxMDljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDA5MDEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiNWRlMThkODFhY2ZlZjlkNDEzYWE1NDhlMjllOTY4ODM1OWQ3ODc1OWEwZDdiOWI4ZGU5NDQ2ODU3YTE5YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XJPliWPs5X9J4uc4CqhL6SnhoC5jLXb2fvcOZdcF6wM"> note: oxygen support is not a Ventilator Interface
> 
> I believe I missed the changes when merging the translations; I'll get it fixed 👍
> 
> As far as this goes,
> 
> > still oxygen support is showing under as "ventilator Interface" on events module
> 
> Is it acceptable to have "Respiratory support" as the caption for oxygen support as well as NIV/IV? @samholics

yes we can do  " Respiratory Support Type"

Jacobjeevan commented:
> yes we can do " Respiratory Support Type"

👍 

<img width="411" alt="image" src="https://github.com/user-attachments/assets/17a5b11a-0d3d-4872-9f58-8f9c705ad3e2">

<img width="411" alt="image" src="https://github.com/user-attachments/assets/672b22cf-78f0-48bf-8717-7ad5e46610d2">

Jacobjeevan committed:
log fixes

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

samholics commented:
> > yes we can do " Respiratory Support Type"
> 
> 👍
> 
> <img alt="image" width="411" src="https://private-user-images.githubusercontent.com/40040905/378370268-17a5b11a-0d3d-4872-9f58-8f9c705ad3e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MzE3MzIsIm5iZiI6MTcyOTUzMTQzMiwicGF0aCI6Ii80MDA0MDkwNS8zNzgzNzAyNjgtMTdhNWIxMWEtMGQzZC00ODcyLTlmNTgtOGY5YzcwNWFkM2UyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDE3MjM1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlNTQxZjA2ZTYzMDgxZTM1MmMzMWYxNzJkMGU3OTFkMWE0MjBhZWU1YjI4NmVmZGI0Mzg4OTJmNGVjY2FkMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t9NBgNTCKq4WUgbhVUvJlYdwJVbOz04sR2qhzihtEps"> <img alt="image" width="411" src="https://private-user-images.githubusercontent.com/40040905/378370368-672b22cf-78f0-48bf-8717-7ad5e46610d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MzE3MzIsIm5iZiI6MTcyOTUzMTQzMiwicGF0aCI6Ii80MDA0MDkwNS8zNzgzNzAzNjgtNjcyYjIyY2YtNzhmMC00OGJmLTg3MTctN2FkNWU0NjYxMGQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDE3MjM1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRiMDVjYTRmZmM2ZjExNmY4M2FiOTVhN2FkNzk5YWY2MTNiYmJhNzdkYjE5ZjdiNTE1NjIxOGY4MGNkYTIwMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Er63eJQM3P6vcyqlSlu_jRKa3bDcRwuNwrFcFn2Y-Ys">

- [ ] Rename INVASIVE to INVASIVE VENTILATOR and Non-INVASIVE to NON-INVASIVE VENTILATOR on Respiratory support type
invasive or non-invasive alone doesn't make sense when its captioned under Respiratory Support type.

Jacobjeevan committed:
IV/NIV log text modification

Jacobjeevan commented:
>     * [ ]  Rename INVASIVE to INVASIVE VENTILATOR and Non-INVASIVE to NON-INVASIVE VENTILATOR on Respiratory support type
>       invasive or non-invasive alone doesn't make sense when its captioned under Respiratory Support type.

👍 

<img width="424" alt="image" src="https://github.com/user-attachments/assets/91ee959d-0e6f-4e0c-862d-c2eadec9cce8">

samholics commented:
> > ```
> > * [ ]  Rename INVASIVE to INVASIVE VENTILATOR and Non-INVASIVE to NON-INVASIVE VENTILATOR on Respiratory support type
> >   invasive or non-invasive alone doesn't make sense when its captioned under Respiratory Support type.
> > ```
> 
> 👍
> 
> <img alt="image" width="424" src="https://private-user-images.githubusercontent.com/40040905/378680687-91ee959d-0e6f-4e0c-862d-c2eadec9cce8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NzkzMDIsIm5iZiI6MTcyOTU3OTAwMiwicGF0aCI6Ii80MDA0MDkwNS8zNzg2ODA2ODctOTFlZTk1OWQtMGU2Zi00ZTBjLTg2MmQtYzJlYWRlYzljY2U4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA2MzY0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3YTdkOTdjZjA3NzUwZmQ1NWJiYjNjZDhkZTZjMTMxMjIzY2U3NWJiM2EzODNkZmI3MTE1N2ZlYWVjOTRjNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TrBA-e5DPlePuAhQsg8scduJzbRYxFvBdEDjkypOGPk">

Tested good to go @nihal467

nihal467 commented:
![image](https://github.com/user-attachments/assets/5b040866-cc32-45a6-a79d-af5e79232372)

@Jacobjeevan After I created a detailed log update, I submitted a brief log update for the patient. This resulted in the removal of the respiratory support badge from the patient consultation section. However, despite the removal of the respiratory badges, the ventilator log's end date and time were not updated. Can you explain how the ventilator end-time logic works?

Jacobjeevan commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/379130693-5b040866-cc32-45a6-a79d-af5e79232372.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2NjgyMjAsIm5iZiI6MTcyOTY2NzkyMCwicGF0aCI6Ii81NzA1NTk5OC8zNzkxMzA2OTMtNWIwNDA4NjYtY2MzMi00NWE2LWE3OWQtYWY1ZTc5MjMyMzcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDA3MTg0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiM2RkMzZlMjhmYjZjNzhkMTA3ZWU0NDVjYjYwZTk1ODE5M2E3NGQzODIwMmExNTM1MTY5YTE2NDA4OTFhNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vwydRvdIHZflstOzHMG29ABVOgV2UO8zr4Tc72dJSjA)
> 
> @Jacobjeevan After I created a detailed log update, I submitted a brief log update for the patient. This resulted in the removal of the respiratory support badge from the patient consultation section. However, despite the removal of the respiratory badges, the ventilator log's end date and time were not updated. Can you explain how the ventilator end-time logic works?

End time is set to the `taken_at` of the next daily round, however I just checked and I'm filtering for rounds where ventilator interface is not null earlier in the code - which is causing this issue. I'll get it fixed 👍

Jacobjeevan committed:
Fixes and cleanup

- Switched VentilatorPlot to use getDailyReports (instead of getDailyRoundsAnalyse).
- Changed end_date to be based on next daily round (rather than round with ventilator data)
- Pagination on top level (in the tab, instead of in Plots)
 - Applies to both Table and the Graphs

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

Jacobjeevan commented:
To offer an update, while working on Vitals redudant api call issue, I realized we could just use useDailyReports route at top level and pass in the data to both the table and VentilatorPlot (right now, VentilatorPlot uses DailyReportsAnalyse route).

So finalizing those changes and testing right now 👍

Jacobjeevan committed:
Cleanup

- Removed filtering items in VentorTable beforehand
	- allowing to fetch the right end date
	- as well as only combining rounds with same NIV/IV/Oxygen mode/modality if there are consecutive (i.e. no round in between them)
- VentilatorPlot
	- adjust label position for the first data point (to avoid overflow of line label into y axis labels)

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

Jacobjeevan committed:
Pagination Limit

- Added limit of 36 (same as before for pagination in VentilatorPlot)

nihal467 commented:
LGTM

rithviknishad commented on src/components/Facility/ConsultationDetails/Events/EventsList.tsx:
```diff
@@ -64,6 +64,13 @@ export default function EventsList({ query }: { query: QueryParams }) {
 
                       const values = Object.fromEntries(entries);
+                      if (
+                        values.ventilator_interface &&
```
nitpicks: this is no-op right

```suggestion
```

Jacobjeevan replied:
Yea 😆

rithviknishad commented on src/components/Facility/Consultations/VentilatorTable.tsx:
```diff
@@ -0,0 +1,124 @@
+  const { dailyRoundsList } = props;
+  let sortedData: DailyRoundsModel[] = [];
+
+  const sortData = (data: DailyRoundsModel[]) => {
+    return data.sort(function (a, b) {
+      const ad = new Date(a.taken_at ?? Date.now());
+      const bd = new Date(b.taken_at ?? Date.now());
+      return ad > bd ? 1 : -1;
+    });
+  };
```
checkout the utility fn: `compareBy` or `compareByDateString` right. we could re-use that here.

https://github.com/ohcnetwork/care_fe/blob/develop/src/Utils/utils.ts#L416-L428

shivankacker commented on src/components/Facility/Consultations/VentilatorTable.tsx:
```diff
@@ -0,0 +1,124 @@
+    return modeOrModality;
+  };
+
+  const VentilatorTableBody = (dailyRoundsList: DailyRoundsModel[]) => {
+    const rows = [];
+    for (let index = 0; index < dailyRoundsList.length; index++) {
```
a map should be more efficient and easy to use than a for loop here

Jacobjeevan replied:
True, though within this loop, I'm traversing/skipping consecutive rounds (the while loop) that have same interface and mode/modality. That should be more readable with a for loop, no? 🤔  @shivankacker 

Same for below.

shivankacker replied:
Sorry just saw this. If it seems more readable to you this way sure

shivankacker commented on src/components/Facility/Consultations/VentilatorPlot.tsx:
```diff
@@ -28,44 +26,194 @@ const modality: Array<ModalityType> = [
+  const getMarkLineData = (name: string) => {
+    const markLineData = [];
+    if (!dailyRoundsList) return [];
+    for (let index = 0; index < dailyRoundsList.length; index++) {
```
same here

Jacobjeevan committed:
using date utils

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

bodhish commented:
@coderabbitai full review

Jacobjeevan commented on src/components/Facility/ConsultationDetails/ConsultationVentilatorTab.tsx:
```diff
@@ -1,21 +1,43 @@
+  const { consultationId } = props;
+  const { qParams, Pagination, resultsPerPage } = useFilters({ limit: 36 });
+
+  const { loading: isLoading, data } = useQuery(routes.getDailyReports, {
+    pathParams: { consultationId },
+    query: {
+      limit: resultsPerPage,
+      offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
+    },
+  });
```
Not needed as children are handling the case where data is empty/undefined.

Jacobjeevan commented on src/components/Facility/ConsultationDetails/Events/EventsList.tsx:
```diff
@@ -64,6 +64,12 @@ export default function EventsList({ query }: { query: QueryParams }) {
                       }
 
                       const values = Object.fromEntries(entries);
+                      if (
+                        values.ventilator_interface === "INVASIVE" ||
+                        values.ventilator_interface === "NON_INVASIVE"
+                      ) {
+                        values.ventilator_interface += " VENTILATOR";
+                      }
```
This is related to displaying the values as per the requirement in daily logs and is needed.

Jacobjeevan committed:
improvement suggestions

Jacobjeevan commented on src/components/Facility/Consultations/VentilatorPlot.tsx:
```diff
@@ -28,44 +26,194 @@ const modality: Array<ModalityType> = [
+      graphData = dailyRoundsData.reduce(
+        (acc, currentRound: DailyRoundsModel) => ({
+          ...acc,
+          //@ts-expect-error taken_at should always be available
```
Don't want to update the interface at the moment.

Jacobjeevan committed:
Nitpick fixes

Jacobjeevan commented on src/components/Facility/Consultations/VentilatorPlot.tsx:
```diff
@@ -28,44 +26,196 @@
+          ? round.ventilator_oxygen_modality
+          : null;
+    return modeOrModality;
+  };
```
Switching to case statements here.

Jacobjeevan commented on src/components/Facility/Consultations/VentilatorPlot.tsx:
```diff
@@ -28,44 +26,196 @@
-    };
+    }
+    return { graphData, graphDataCount };
+  };
```
Interface added for graph results.

Jacobjeevan committed:
type fix

Jacobjeevan commented on src/components/Facility/Consultations/components/BinaryChronologicalChart.tsx:
```diff
@@ -3,7 +3,7 @@ import { formatDateTime } from "../../../../Utils/utils";
 export default function BinaryChronologicalChart(props: {
   data: {
-    value: boolean;
+    value: boolean | undefined;
```
Not necessary as earlier checks would prevent such a scenario (additionally, timestamp should have values if there's daily rounds data).

Jacobjeevan committed:
Merge branch 'develop' into issues/8264-VentilatorData-ConsultationPage

Jacobjeevan committed:
lint fix

Jacobjeevan commented on src/components/Facility/ConsultationDetails/ConsultationVentilatorTab.tsx:
```diff
@@ -1,20 +1,44 @@
 
 export const ConsultationVentilatorTab = (props: ConsultationTabProps) => {
+  const { consultationId } = props;
+  const { qParams, Pagination, resultsPerPage } = useFilters({ limit: 36 });
```
It's 36 daily log events and not every event would have a corresponding ventilator table entry.
